File: modules/09-reflection/kotlin/app/Main.kt
// Shows the generated members: toString, equals, and copy. Deterministic.
fun main() {
    val a = Person("Ada", 36)
    println("toString (generated): $a")
    println("equals (generated):   ${a == Person("Ada", 36)}")
    val b = a.copy(age = 37) // copy (generated)
    println("copy (generated):     $b")
    println("not equal after copy: ${a == b}")
}

File: modules/09-reflection/ocaml/bin/main.ml
(* Shows the hand-written generic operations. Deterministic. *)
let () =
  let open Codegen in
  let a = { name = "Ada"; age = 36 } in
  Printf.printf "to_json (hand-written): %s\n" (to_json a);
  Printf.printf "show (hand-written):    %s\n" (show a);
  Printf.printf "equal (hand-written):   %b\n" (equal a { name = "Ada"; age = 36 });
  Printf.printf "not equal after edit:   %b\n" (equal a { name = "Ada"; age = 37 })
